dev/core/src/com/google/gwt/dev/jjs/impl/ImplementJsVarargs.java

```diff
@@ -462,8 +462,8 @@ public void endVisit(JMethodCall x, Context ctx) {
       // Passed as an array to varargs method will result in an apply call, in which case hoist the
       // qualifier to make sure it is only evaluated once.
       JExpression instance = x.getInstance();
-      if (x.getTarget().needsDynamicDispatch() && !x.isStaticDispatchOnly()
-          && instance != null && !(instance instanceof JVariableRef)) {
+      if (x.getTarget().needsDynamicDispatch() && !x.isStaticDispatchOnly() && instance != null
```

Comment:
Since we dropped support for java 7 and now we only support java 8+ wont it make sense to start using newer syntax like Objects.nonNull(...) and Objects.isNull(...)

Reply:
Specifically, we dropped support for _running this test_ on Java 7 - the code may or may not still run, but the test won't pass. We are not yet compiling code to use Java 9+ APIs, but could use Java features like that.

That said, two thoughts:
 * isNull() is great to use as a lambda, but it any better than the != null? It appears to be the same number of chars (assuming you add spaces, and assuming you do a static import), but will add the cost of a method call (until it is inlined by the jit, etc). As I understand it, the purpose is mostly for passing as a method reference...?
  * If I'm not changing the logic (I only moved the null check up a line), I'd rather not change the code to another way of doing the same thing, all else equals.